rubix-core/src/main/java/com/qubole/rubix/core/CachingFileSystem.java

```diff
@@ -319,6 +321,33 @@ private URI getOriginalURI(URI actualURI)
     return URI.create(getScheme() + "://" + actualURI.getAuthority());
   }
 
+  @Override
+  public RemoteIterator<LocatedFileStatus> listLocatedStatus(Path path)
+          throws IOException
+  {
+      return new RemoteIterator<LocatedFileStatus>()
+      {
+          private final RemoteIterator<LocatedFileStatus> stats = fs.listLocatedStatus(path);
```

Comment:
also, any reasons for not doing `listStatus`? we're anyways not using the block locations returned.

Reply:
> any reasons for not doing listStatus? we're anyways not using the block locations returned.

List status fetches all statuses at once. This could be problematic if directory has thousands of files (or you don't need to read all files). E.g Presto uses `listLocatedStatus` in an iterative way to to provide splits.

Reply:
ok, that's makes sense. 👍